pandas/tests/plotting/test_frame.py

```diff
@@ -139,6 +139,23 @@ def test_plot(self):
             result = ax.get_axes()  # deprecated
         assert result is axes[0]
 
+    # GH 18726
+    def test_two_plots_with_x_str(self):
+        x1, y1 = ['a', 'b'], [1, 2]
+        x2, y2 = ['b', 'a'], [4, 3]
+        df1 = pd.DataFrame(y1, index=x1)
+        df2 = pd.DataFrame(y2, index=x2)
+        ax = None
+        ax = df1.plot(ax=ax)
+        ax = df2.plot(ax=ax)
+
+        line1, line2 = ax.lines
+        # xdata should not be touched (Earlier it was [0, 1])
+        np.testing.assert_equal(line1.get_xdata(), x1)
```

Comment:
we don't use ``np.testing``, instead use ``tm.assert_numpy_array_equal``

Reply:
sure it does
turn the list into a numpy array
you can not compare a list and an array in our testing by default

Reply:
ok, I made that change